src/main/java/com/testlio/plugins/jenkins/helper/PollResultHelper.java

```diff
@@ -18,15 +21,24 @@ public static void pollResults(TaskListener listener, RestClient restClient, Run
       return;
     }
 
-    listener.getLogger().println("Step 8.1: Initially Waiting for: "+INITIAL_INTERVAL/1000 +" seconds, before result pooling starts");
+    listener.getLogger().println("Step 8.1: Initially Waiting for: " + INITIAL_INTERVAL / 1000 + " seconds, before result polling starts");
     Thread.sleep(INITIAL_INTERVAL);
 
     final String automatedRunResultHref = "/automated-test-run/v1/collections/" + config.getAutomatedTestRunCollectionGuid() + "/results-new?testRunHref=" + automatedRun.getHref();
-    String resultStatus = null;
+    String resultStatus = "";
     int failedRequests = 0;
     listener.getLogger().println("Step 8.2: Starting to poll for results at an interval of " + POLLING_INTERVAL / 1000 + " seconds.");
-    while (failedRequests <= MAX_FAILED_POLLS) {
-      AutomatedRunResultsDTO resultResponse = (AutomatedRunResultsDTO) restClient.get(automatedRunResultHref, AutomatedRunResultsDTO.class);
+    while (StringUtils.isNotBlank(resultStatus)) {
+      ResponseEntity<String> response = restClient.getWithResponseEntity(automatedRunResultHref);
+
+      listener.getLogger().println("ResponseCode:" + response.getStatusCode());
+      if (response.getStatusCode() != HttpStatus.OK) {
+        if (failedRequests >= MAX_FAILED_POLLS) {
+          listener.getLogger().println("Failed to poll results, Trying again");
+        }
```

Comment:
Should throw an error here and abort the process with a failure, otherwise there's no point in having a max failed polls threshold

Reply:
Yes, you are right. Incorporated the comments and tested it